src/components/atomic-swap/ExternalChangeModal.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from "react";
+import { t } from "ttag";
+import $ from 'jquery';
+
+export function ModalAtomicExternalChange ({ manageDomLifecycle }) {
+    const modalDomId = 'atomicExternalChangeModal';
+
+    useEffect(() => {
+        manageDomLifecycle(`#${modalDomId}`);
+    }, [])
+
+    return <div><div className="modal fade"
```

Comment:
What happens if you are signing a tx and right before signing the proposal is updated with a different output? Are you going to sign it? The action is cancelled?

Reply:
The biggest probability is that the action is cancelled.

However on the backend side only one of them will be able to execute the writing, since the query to update the proposal includes the version number. The other will receive a `409 Conflict` error and will have to re-do its actions.

Reply:
Adding to this comment:
- The most common case will be the user receiving a realtime notification that the proposal was changed elsewhere and losing its changes since the last "Save and Update" click
- The second most probable case will be a race condition in which the other participant makes the change to the proposal but the message doesn't have enough time to reach the current user. The request signing is then made, but once sent to the backend receives the `409 conflict` error
- In any case, the added signatures are voided since the PartialTx is modified. There is no way to inadvertently sign a transaction without a very clear message from the interface that the transaction has changed.

The `Summary` table at the top of the `EditSwap` screen is designed to help identify these changes easily and prevent any malicious actions.